### PR TITLE
Add reusable workflow for backing up repo

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: .github/workflows/backup-repo.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  backup_repo_to_aws_codecommit:
+    name: AWS CodeCommit
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+        with:
+          fetch-depth: 0 # Fetch the entire history to backup everyting
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply .github/workflows/backup-repo.yml:18]: It is

File: .github/workflows/backup-repo.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  backup_repo_to_aws_codecommit:
+    name: AWS CodeCommit
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply .github/workflows/backup-repo.yml:11]: Might make more sense to have that in https://github.com/exercism/org-wide-files/pull/304/files?

File: .github/workflows/backup-repo.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        run: |
+          mkdir -p ~/.ssh
+          echo "${{ secrets.AWS_CODECOMMIT_SSH_PRIVATE_KEY }}" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+          ssh-keyscan -t rsa git-codecommit.${{ secrets.AWS_REGION }}.amazonaws.com >> ~/.ssh/known_hosts
+
+      - name: Push repo to AWS CodeCommit
+        run: |
+          git remote add aws-codecommit "ssh://${{ secrets.AWS_CODECOMMIT_SSH_PRIVATE_KEY_ID }}@git-codecommit.${{ secrets.AWS_REGION }}.amazonaws.com/v1/repos/${{ github.event.repository.name }}"
+          git push --tags --force --prune aws-codecommit "refs/remotes/origin/*:refs/heads/*"
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply .github/workflows/backup-repo.yml:30]: 🤷 But if we don't do `--force`, won't it break when someone does force push (happens rarely, but it does happen)?

File: .github/workflows/backup-repo.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+        with:
+          fetch-depth: 0 # Fetch the entire history to backup everyting
+
+      - name: Setup SSH for AWS CodeCommit
+        run: |
+          mkdir -p ~/.ssh
+          echo "${{ secrets.AWS_CODECOMMIT_SSH_PRIVATE_KEY }}" > ~/.ssh/id_rsa
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply .github/workflows/backup-repo.yml:23]: Is there any other way we can do this?

File: .github/workflows/backup-repo.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply .github/workflows/backup-repo.yml:11]: Done

File: .github/workflows/backup-repo.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply .github/workflows/backup-repo.yml]: I've removed `--force`

File: .github/workflows/backup-repo.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (iHiD) [reply .github/workflows/backup-repo.yml:23]: > Does this mean that every user with write access to an Exercism repo has the ability to read this private key if they want to? And if so, do we consider that a problem?

Is this same question not relevant to any secret?

[the remark this answers is not in the document]
Comment (iHiD) [reply .github/workflows/backup-repo.yml:23]: > I don't know what the AWS_CODECOMMIT_SSH_PRIVATE_KEY credential allows. Can it basically only push new backups, or can it also delete old backups and change CodeCommit settings?

It has read/write permissions to git. No permissions beyond that.

File: .github/workflows/backup-repo.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply .github/workflows/backup-repo.yml]: (You usually want force with leash in cases like this so the case where someone else has comitted inbetween cannot force push replace the code)

**Pull request closed without merging**